sympy/polys/matrices/tests/test_ddm.py

```diff
@@ -1,12 +1,16 @@
 from sympy.testing.pytest import raises
 from sympy.external.gmpy import GROUND_TYPES
 
+import pytest
 from sympy.polys import ZZ, QQ
 
 from sympy.polys.matrices.ddm import DDM
 from sympy.polys.matrices.exceptions import (
     DMShapeError, DMNonInvertibleMatrixError, DMDomainError,
     DMBadInputError)
+from sympy.polys.matrices import DomainMatrix
+DMZ_all = [DomainMatrix]
+DMQ_all = [DomainMatrix]
 
```

Comment:
I didn't mean to repeat this here. Rather there should be tests for the `.qr()` method in the `test_xxm.py` test file. That way the test can cover all of `DDM`, `SDM` etc. Since we have several types that are supposed to have the same interface it is better if they can share their tests.